### ISSUE TITLE
Dockerfile?

### ISSUE THREAD

**Issue by tomasz-nolberczak** [jstolarek/dc-card-maker-script#1]
Hi, thanks for good work! 👏

Have a question about Docker support. Are there any chances we could run that script within Docker image?

**Comment by jstolarek** [comment]
I honestly have no idea, I'm not familiar with Docker at all. If Docker can read from and write to system directories outside of the image itself then yes, this should be possible.

**Comment by tomasz-nolberczak** [comment]
Yes, Docker can read from and write to system directories. I am gonna try to add a support for it in near future. Will do a PR then.

**Comment by jstolarek** [comment]
After sleeping on this I think that I'm not comfortable about accepting code that I'm later unable to maintain. (That, and the fact that creating a docker image for a single bash script feels like an overkill.) My suggestion is that you create a separate project for the docker file and add my repository as a git submodule. Once it's done I'll link to your docker project from the readme. How does that sound?

**Comment by tomasz-nolberczak** [comment]
> creating a docker image for a single bash script feels like an overkill.

For me, creating a Dockerfile isn't an overkill. Maybe you ask me why? I feel that someone could want to keep his system clean. Installing few specific modules for only one tool might be hard for someone, especially for people who don't have a time to dig how to install them. Having everything in one Dockerfile might improve their experience with a script and can encourage them to use it.

> My suggestion is that you create a separate project for the docker file and add my repository as a git submodule. 

Sounds nice! I see your point and agree with that. 

Right now I don't have much time to do that, however in next few days I will get the GDEMU module, so then there will be a need to create the SD card with a lot of games. Will notify you here in this discussion about the progress.

**Comment by jstolarek** [comment]
> For me, creating a Dockerfile isn't an overkill.

I understand that the script itself will be small. What I meant by overkill is having a container for a single bash script. But of course, there are two dependencies to be compiled from source. I'd argue that if you're using Linux as your primary OS and have a Dreamcast then it makes sense to have these two tools on your system, but to each their own I suppose.

Interestingly, `cdirip` sources already come with files necessary to build a Debian package. I wonder what would it take to get it into the Debian repositories.

Anyway, there's no rush. Once it's done I'll just update the readme. Also, I'm not sure whether my submodule idea makes sense in the context of Docker. You can probably just point to my repo as an upstream, right?
